crates/matrix-sdk-appservice/src/lib.rs

```diff
@@ -545,7 +545,18 @@ mod tests {
     use super::*;
 
     fn registration_string() -> String {
-        include_str!("../tests/registration.yaml").to_owned()
+        r"
+             id: appservice
+             url: http://127.0.0.1:9009
+             as_token: as_token
+             hs_token: hs_token
+             sender_localpart: _appservice
+             namespaces:
+               users:
+               - exclusive: true
+                 regex: '@_appservice_.*'
+         "
```

Comment:
Shouldn't this use something like [indoc](https://docs.rs/indoc/latest/indoc/index.html)'s `formatdoc!` to strip all the leading whitespace?

Reply:
It's correctly parsed by `serde_yaml`. I wonder what the yaml spec says about indentation. Given this is the first example that consumsers see, it should maybe actually use the file-based registration instead of string-based one, as the latter is supposed to be used only for development purposes.

Reply:
As an aside, going by the spec (https://yaml.org/spec/1.2.2/#61-indentation-spaces) leading whitespaces for top-level keys are fine.

Reply:
Right. Do you still want to do the change to have the example read the registration from a file, or should we merge as-is?

Reply:
Changed to reading from file.